### PR TITLE
Add nginx check plugin

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: collectd/files/plugin/collectd_nginx_check.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        super(NginxCheckPlugin, self).__init__(*args, **kwargs)
+        self.plugin = NAME
+        self.url = None
+
+    def config_callback(self, conf):
+        super(NginxCheckPlugin, self).config_callback(conf)
+
+        for node in conf.children:
+            if node.key == 'Url':
+                self.url = node.values[0]
```
Comment (elemoine) [inline collectd/files/plugin/collectd_nginx_check.py:35]: You may want to `break` when a node with a URL was found.

**Review by elemoine — CHANGES_REQUESTED** [review]
Minor comment, otherwise looks good to me.

File: collectd/files/plugin/collectd_nginx_check.py (hunk 1 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_nginx_check.py:35]: Done

**Review by elemoine — APPROVED** [review]

File: collectd/files/plugin/collectd_nginx_check.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                break
+
+        if self.url is None:
+            self.logger.error("{}: Missing Url parameter".format(NAME))
+
+    def read_callback(self):
+        try:
+            requests.get(self.url, timeout=5)
```
Comment (SwannCroiset) [inline collectd/files/plugin/collectd_nginx_check.py]: timeout is a base configuration (handled by base.Base) , you can use self.timeout in place to hardcoded value.

Comment (SwannCroiset) [inline collectd/files/plugin/collectd_nginx_check.py]: IIRC, the requests.get doesn't return necessary an exception, you have to check the HTTP return code

**Review by SwannCroiset — CHANGES_REQUESTED** [review]
Why not re-using the HTTPCheckPlugin ?

File: collectd/files/plugin/collectd_nginx_check.py (hunk 2 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_nginx_check.py]: TBH I've adapted the already existing Apache check plugin which indeed has all the shortcomings that you describe.

**Comment by simonpasquier** [comment]
@SwannCroiset 

> Why not re-using the HTTPCheckPlugin ?

right now the data collected by HTTPCheckPlugin translate into check_local_endpoint metrics. Eventually it should be generic but this requires additional work.

File: collectd/files/plugin/collectd_nginx_check.py (hunk 2 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_nginx_check.py]: also I think it's fine for now that the plugin doesn't check the return code. As long as the server replies, the check metric can be OK.

**Review by SwannCroiset — APPROVED** [review]
ok lets merge it as is

**Pull request merged**